pkg/apis/extensions/types.go

```diff
@@ -232,6 +232,26 @@ type DeploymentSpec struct {
 	// Indicates that the deployment is paused and will not be processed by the
 	// deployment controller.
 	Paused bool `json:"paused,omitempty"`
+	// The config this deployment is rolling back to. Will be cleared after rollback is done.
+	RollbackTo *RollbackConfig `json:"rollbackTo,omitempty"`
+}
+
+// DeploymentRollback stores the information required to rollback a deployment.
+type DeploymentRollback struct {
+	unversioned.TypeMeta `json:",inline"`
+	// Required: This must match the Name of a deployment.
+	Name string `json:"name"`
+	// Required: This must match the Namespace of a deployment.
+	Namespace string `json:"namespace"`
```

Comment:
Discussed offline: Namespace is in api.Context, so we apparently don't need it in subresources, in general.